model/prelude_mem_addrtype.sail

```diff
@@ -0,0 +1,14 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+newtype physaddr = physaddr : xlenbits
+newtype virtaddr = virtaddr : xlenbits
```

Comment:
Can you add a note:

> Note virtaddr includes physical addresses when address translation is disabled. It is used where the address would be virtual if address translation is active.

It's slightly confusing otherwise and I don't think we're going to come up with a clearer name (`untranslatedaddr`?...)

Reply:
I find this comment confusing and I believe this is not needed? 

We already have a "no-op" translation from virtaddr to physaddr for the bare case (`Sbare => return TR_Address(physaddr(virtaddr_bits(vAddr)), init_ext_ptw),`) so we can just always start out with a virtual address, but in the SBare case it just happens to be 1:1 mapped to a physical one?

Reply:
However, in chip implementations that do not support virtual addresses, there is no concept of a virtual address, so virtaddr should be interpreted as physaddr, rather than through translation. The Sbare implementation just happens to be compatible with this situation while also being compatible with direct read and write of physical memory in M-mode

Reply:
Yeah exactly. Also in `Sbare` mode I think it's natural to think of the address as physical addresses rather than virtual addresses that go through an identity translation.

Maybe my comment didn't explain that very well, but I think we need to have some kind of explanation of this subtlety.

Reply:
If you would like to define it that way, I would prefer to add something like a "cpuaddr" that is explicitly converted to virtaddr/physaddr depending on whether a MMU is present/enabled. That is IMO cleaner than saying virtaddr can be physaddr depending on where it is being used. For a m-mode only CPU I'd be fine with cpuaddr being physaddr, but I still view sbare as a 1:1 mapping rather than addresses being physical when switching off the MMU (since the MMU could be turned on again).

Once hypervisor is implemented we should add intermediate physical addresses that are stage2 translated before being turned into physaddrs, this also means directly using physaddr when the stage1 MMU is disabled is not correct.